src/main/java/org/apache/mesos/scheduler/recovery/constrain/TimedLaunchConstrainer.java

```diff
@@ -10,29 +10,33 @@
  * rate-limiting purposes.
  */
 public class TimedLaunchConstrainer implements LaunchConstrainer {
-    private AtomicLong lastLaunch;
+    private AtomicLong lastPermanentRecoveryLaunch;
     private long minDelaySeconds;
 
     /**
-     * Create a new constrainer with the given required minimum delay between recovery operations.
+     * Create a new constrainer with the given required minimum delay between permanent (destructive) recovery
+     * operations.
      *
      * @param minDelaySeconds Minimum number of seconds between each launch
      */
     public TimedLaunchConstrainer(long minDelaySeconds) {
         this.minDelaySeconds = minDelaySeconds;
-        this.lastLaunch = new AtomicLong(0);
+        this.lastPermanentRecoveryLaunch = new AtomicLong(0);
     }
 
     @Override
-    public void launchHappened(Operation launchOperation) {
-        if (!canLaunch(null)) {
-            throw new RuntimeException("It hasn't been long enough since the last launch");
+    public void launchHappened(Operation launchOperation, RecoveryRequirement.RecoveryType recoveryType) {
+        if (recoveryType.equals(RecoveryRequirement.RecoveryType.Permanent)) {
+            lastPermanentRecoveryLaunch.compareAndSet(lastPermanentRecoveryLaunch.get(), System.currentTimeMillis());
         }
-        lastLaunch.compareAndSet(lastLaunch.get(), System.currentTimeMillis());
     }
 
     @Override
-    public boolean canLaunch(OfferRequirement offerRequirement) {
-        return (lastLaunch.get() + minDelaySeconds * 1000) < System.currentTimeMillis();
+    public boolean canLaunch(RecoveryRequirement recoveryRequirement) {
+        if (recoveryRequirement.getRecoveryType().equals(RecoveryRequirement.RecoveryType.Permanent)) {
```

Comment:
Could add static utility methods to RecoveryRequirement of this form:
- public static boolean isPermanent(RecoveryRequirement)
- public static booleas isTemporary(RecoveryRequirement)

Similar to (but the opposite of) Block.getStatus(): https://github.com/mesosphere/dcos-commons/blob/3581e7b5697d9789c1162200398637186df82423/src/main/java/org/apache/mesos/scheduler/plan/Block.java